sys/include/memarray.h

```diff
@@ -73,7 +84,14 @@ void *memarray_alloc(memarray_t *mem);
  * @return pointer to allocated structure, if enough memory was available
  * @return NULL, on failure
  */
-void *memarray_calloc(memarray_t *mem);
+static inline void *memarray_calloc(memarray_t *mem)
+{
+    void *new = memarray_alloc(mem);
+    if (new) {
+        memset(new, 0, mem->size);
```

Comment:
This might also be part of the secret sauce reducing the ROM footprint: Modern compilers understand the semantics of `memset()` and friends and only emit a function call, if this is indeed better. Especially if the memory is initialized later on anyway, the `memset()` is identified as a ddead write and fully optimized out.

Reply:
I mean: By `inline`-ing the `malloc()`, the compiler can indeed optimize this due to the context being available.

Reply:
I also assume that when using the `calloc` version, the `if`-statement in `calloc` and `alloc` can be merged by the compiler into a single check.